### PR TITLE
make sure scopes are added to redirectURL

### PR DESCRIPTION
**Pull request by dalbrx-forcam**
Fixes a bug with OAuth2 scopes

When debugging we found out that there is an Array.isArray(scopes) call which fails because scopes is of type ImmutableList, so no scopes are added to the redirect URI. 

see https://github.com/swagger-api/swagger-ui/blob/139592e353f47ef4f9c567a36624da386c661f17/src/core/oauth2-authorize.js#L55

### Description

Check if scopes has a toArray method and use the result of it if Array.isArray(scopes) returns false.

### Motivation and Context
Fixed the following issue (when using implicit grant):

1. Click on Authorize Button
2. Select all available scopes
3. Click on Popup Authorize Button (redirect URI does contain scopes)
4. Call any API
5. Cick on Authorize Button
6. Deselect all available scopes
7. Click on Popup Authorize Button (redirect URI does not contain scopes)
8. Call any API
9. Click on Authroize Button
10. Select all available scopes
11. Click on Popup Authorize Button (**redirect URI does not contain scopes!**)

### How Has This Been Tested?

Tested the above steps manually.

### Screenshots (if appropriate):

The following screenshot shows the object which fails the Array.isArray check:

swagger-scopes

### My PR contains... 
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [X] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [X] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/oauth2-authorize.js [hunk 1]
```diff
@@ -52,10 +52,17 @@ export default function authorize ( { auth, authActions, errActions, configs, au
   }
   query.push("redirect_uri=" + encodeURIComponent(redirectUrl))
 
-  if (Array.isArray(scopes) && 0 < scopes.length) {
+  let scopesArray = []
+  if (Array.isArray(scopes)) {
+    scopesArray = scopes
+  } else if (typeof scopes !== "undefined" && typeof scopes.toArray === "function") {
```
Comment (tim-lai) [inline src/core/oauth2-authorize.js:58]: Immutable has its own isType() static function which is more readable. Suggest importing Im, then change to `if (Im.List.isList(scopes)) { do something}`

File: src/core/oauth2-authorize.js [hunk 2]
```diff
@@ -52,10 +52,17 @@ export default function authorize ( { auth, authActions, errActions, configs, au
...
+    scopesArray = scopes.toArray()
+  }
+
+  if (0 < scopesArray.length)  {
```
Comment (tim-lai) [inline src/core/oauth2-authorize.js:62]: legacy, but should use common conventions... `if (scopes.Array.length > 0) { do something }

**Review by tim-lai — CHANGES_REQUESTED** [review]
@dalbrx-forcam Thanks for the PR! Requesting some refactoring. Also, any way we can get a Jest or Cypress test to track this?

File: src/core/oauth2-authorize.js (hunk 1 above)
Comment (dalbrx-forcam) [reply src/core/oauth2-authorize.js]: Yes, looks better

File: src/core/oauth2-authorize.js (hunk 2 above)
Comment (dalbrx-forcam) [reply src/core/oauth2-authorize.js]: Sure, I will change this.

**Review by tim-lai — APPROVED** [review]

**Comment by tim-lai** [comment]
@dalbrx-forcam PR merged! Thanks for the contribution!

**Pull request merged**